platform/ui/src/components/measurementTable/MeasurementTable.js

```diff
@@ -75,10 +78,36 @@ class MeasurementTable extends Component {
         <ScrollableArea>
           <div>{this.getMeasurementsGroups()}</div>
         </ScrollableArea>
+        <div className="measurementTableFooter">
+          {saveFunction && (
+            <button onClick={this.saveFunction} className="saveBtn">
+              <Icon name="save" width="14px" height="14px" />
+              Save measurements
+            </button>
+          )}
+        </div>
       </div>
     );
   }
 
+  saveFunction = async event => {
+    const { saveFunction, snackbarContext } = this.props;
+    if (saveFunction) {
+      try {
+        const result = await saveFunction();
+        snackbarContext.show({
+          title: 'STOW SR',
+          message: result.message,
+        });
+      } catch (error) {
+        snackbarContext.show({
+          title: 'STOW SR',
+          message: error.message,
+        });
```

Comment:
Instead of tightly coupling this component to `snackbar`, it should have callbacks for `onError` and `onSuccess` (or something similar. Then, wherever we're using this in `viewer`, should provide the implementation to those callback props. This is a blocker, imo